include/nanobind/nanobind.h

```diff
@@ -51,6 +51,7 @@
 #include "nb_func.h"
 #include "nb_class.h"
 #include "nb_misc.h"
+#include "nb_eval.h"
```

Comment:
Let's make this an optional feature of nanobind. Users will need to include the file `nanobind/eval.h` (so it needs to be renamed and not included from `include/nanobind/nanobind.h`).